torchdynamo support modules() for nn_module

ydwu4 opened the pull request:
Differential Revision: D40820879

This diff allows models to call self.modules() during dynamo tracing.


cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

voznesenskym reviewed (changes requested):
Can you add compile counter to the test? Also, please add a PR description.

ydwu4 committed:
torchdynamo support modules() for nn_module (#88023)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/88023

To enable mt models to call self.modules() in forward.

Test Plan: Test with CIs.

Reviewed By: jansel, tugsbayasgalan

Differential Revision: D40820879

fbshipit-source-id: 988f292097a77df2d540e85b627b6f8d1c7d0a34

ydwu4 commented:
> Can you add compile counter to the test? Also, please add a PR description.

Sure! I've updated the pr accordingly.

voznesenskym commented:
> > Can you add compile counter to the test? Also, please add a PR description.
> 
> Sure! I've updated the pr accordingly.

Beautiful, ty!

ydwu4 commented:
@pytorchbot merge

mehtanirav commented:
@pytorchbot revert -m "[Internal breakages](https://www.internalfb.com/intern/sandcastle/job/13510799692855066/insights)" -c ghfirst

ydwu4 commented:
@pytorchbot merge

ydwu4 commented:
@pytorchbot revert -m "[Internal breakages] https://www.internalfb.com/intern/sandcastle/job/9007200067589062/" -c ghfirst
